### PR TITLE
Updates for new firmware

### LINKED ISSUE #70: Latest firmware update requires Digest auth, breaks HTTP basic auth requests
I'm working with a IP2M-841, just upgraded to new firmware V2.420.AC00.17.R 3/22/17

After upgrading, all my third party tools and custom code using python-amcrest broke on any request:

```
Traceback (most recent call last):
  File "/home/homeassistant/DPtest.py", line 7, in <module>
    print(c.camera.current_time)
  File "/home/homeassistant/.homeassistant/deps/amcrest/system.py", line 19, in current_time
    'global.cgi?action=getCurrentTime'
  File "/home/homeassistant/.homeassistant/deps/amcrest/http.py", line 102, in command
    resp.raise_for_status()
  File "/home/homeassistant/.homeassistant/deps/requests/models.py", line 909, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 401 Client Error: Unauthorized for url: http://xyz.xyz.xyz.xyz:80/cgi-bin/global.cgi?action=getCurrentTime
```

I verified by tcpdump/wireshark that we're sending basic auth in the request, and the camera responds 401 and asks for digest auth.

Seems others are seeing it too:
https://amcrest.com/forum/technical-discussion-f3/cgi-sdk-no-longer-functioning-on-17r--t2401.html

I fixed it by changing to digest auth in http.py:
`self._token = requests.auth.HTTPDigestAuth(self._user, self._password)`

Everything is working now, happy to submit a PR

### PR DESCRIPTION
**Pull request by dparker18**
this fixes #70 and #71 on my recently updated IP2M-841

### CODE REVIEW

File: src/amcrest/http.py [hunk 1]
```diff
@@ -62,7 +62,15 @@ def __init__(self, host, port, user,
         else:
             self._retries_conn = retries_connection
 
-        self._set_name()
+        try:
+            self._set_name()
+        except requests.exceptions.HTTPError as e:
+            # newer firmware requires digest auth, let's try that if we are still unauthorized
```
Comment (tchellomello) [inline src/amcrest/http.py]: Looks good, just make sure the code style is proper within the 79 characters.

File: src/amcrest/system.py [hunk 2]
```diff
@@ -66,7 +66,12 @@ def software_information(self):
         ret = self.command(
             'magicBox.cgi?action=getSoftwareVersion'
         )
-        version, build_date = ret.content.decode('utf-8').split()
+        swinfo = ret.content.decode('utf-8')
+        sep = None
+        if swinfo.find(',') >= 0:
+            swinfo = swinfo.strip() # trailing \r\n
```
Comment (tchellomello) [inline src/amcrest/system.py]: Try to simply this part for example:
```python
swinfo = ret.content.decode('utf-8')
delimiter = ','
if delimiter in swinfo:
    version, build_date = swinfo.split(sep)
else:
   version, build_date = swinfo.split()

**Comment by tchellomello** [comment]
I'm going to merge this and fix the code style on the later commit. 

Thanks for your contribution @dparker18 :+1:

**Comment by dparker18** [comment]
You beat me to it, thank you for fixing the style warnings and merging!

**Pull request merged**